diff --git a/CMakeLists.txt b/CMakeLists.txt
index 3da3f50..b421b98 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -71,10 +70,10 @@ if (CMAKE_CXX_COMPILER_ID STREQUAL "MSVC")

     # C4576: a parenthesized type followed by an initializer list is a non-standard explicit type conversion syntax
     # Used in FFmpeg macros which assumes C.
-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /Zc:preprocessor /wd4068 /wd4576")
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /Zc:preprocessor /Zc:lambda /wd4068 /wd4576")
     add_compile_definitions(_CRT_SECURE_NO_WARNINGS)
     if (BMF_ENABLE_CUDA)
-        set(CMAKE_CUDA_FLAGS "${CMAKE_CUDA_FLAGS} -Xcompiler=/Zc:preprocessor")
+        set(CMAKE_CUDA_FLAGS "${CMAKE_CUDA_FLAGS} -Xcompiler=/Zc:preprocessor -Xcompiler=/Zc:lambda")
     endif()
 endif()

